src/ranch.erl

```diff
@@ -38,15 +38,17 @@
 -export([procs/2]).
 -export([wait_for_connections/3]).
 -export([wait_for_connections/4]).
--export([filter_options/3]).
+-export([filter_options/4]).
 -export([set_option_default/3]).
 -export([require/1]).
 -export([log/4]).
 
 -type max_conns() :: non_neg_integer() | infinity.
 -export_type([max_conns/0]).
 
--type opts() :: any() | #{
+-type opts() :: opts(any()).
+
+-type opts(SocketOpts) :: any() | #{
```

Comment:
I think it would be better to rename the `opts/1` type to `transport_opts/1` and remove the `any()` from it, and include it `opts/0` instead?

From a documentary perspective, it might be even better to write it more verbosely like
```erlang
-type opts() :: socket_opts() | transport_opts(socket_opts()).
-type socket_opts() :: any().
-type transport_opts(SocketOpts) :: #{
        ...
        socket_opts => SocketOpts
}.
```
What do you think?

Reply:
On second thought, that would mean we should rename the `opts/0` types in `ranch_transport`, `ranch_tcp` and `ranch_ssl` to be `socket_opts/0`, too, to be consistent.

Reply:
Depends on whether `opts/1` is public and what it's used for I suppose.

Reply:
Looks fine now.